assignments/this.js

```diff
@@ -1,10 +1,13 @@
-/* The for principles of "this";
-* in your own words. explain the four principle for the "this" keyword below.
+/* The four principles of "this";
+* in your own words, explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. If we don't use any specific rules to point this, this points to the window (when working in the browser).
+
+* 2. This points to whatever is left of the dot.
+
+* 3. Uses .bind() to create a new function.
+
+* 4. In explicit binding, .call() or .apply() are used to determine where this points.
```

Comment:
* 1. If 'this' is not specifically assigned, it is automatically assigned to the window/global object.  This is known as window binding.
* 2. When 'this' is used in object methods then implicit binding is involved and 'this' is automaticaly assigned to an object when called by a preceding dot. So, by calling the objectName.method() the object name immediately to the left of the dot is the one attached to 'this.  If a long list of objectnames: objectName1.objectName2.objectName3.method() - it is the last dot that is the one that counts.
* 3. When a constructor function is actually used, or instantiated, then a 'new' instance of that object can be said to be made.  In these instances when a new object is constructed, or invoked, the 'this' keyword inside the constructor function above is bound to the new object. 
* 4. When 'this' is involved in constructor functions then Explicit binding is involved.  There are 3 ways to attach 'this' to a constructor function: .bind, .call, and .apply